docs/introduction.md

```diff
@@ -0,0 +1,139 @@
+to simplify interactions that otherwise involve repeatedly checking AWS SDK
+documentation.
+
+## What's a `Model`?
+
+A model is a bundle of a human-readable name, some, AWS Batch configuration,
+and an argument validator.
+
+```scala mdoc
+import com.rasterfoundry.granary.datamodel._
+
+import io.circe.syntax._
+
+import java.util.UUID
+
+val model = Model.Create(
+  "A descriptive model name",
+  Validator(().asJson), // a Validator -- more on this below
+  "perfectAccuracyModel:1", // an AWS Batch job definition
+  "veryExpensiveOnDemandQueue" // an AWS Batch job queue
+)
+// What you'll POST to the API to create a model
+model.asJson.spaces2
+```
+
+`model`s in Granary correspond to containers that have been configured to
+run via AWS Batch job definitions. The major difference between using Granary
+and hand rolling `SubmitJob` requests is the `Validator`. To create a model,
+post JSON like what's shown above to `/api/models`.
+
+## What's a `Validator`?
+
+A `Validator` uses JSON Schema to ensure that when you try to run your job,
+you have the correct arguments. For example, in the example above the
+`Validator` is the empty json object `{}`. This means that providing any arguments
+will fail. That model is not a very useful model. We could instead require a green
+band, a red band, and a GeoTIFF location with the following JSON schema:
+
+```json
+{
+  "definitions": {},
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://example.com/root.json",
+  "type": "object",
+  "title": "The Root Schema",
+  "required": [
+    "greenBand",
+    "redBand",
+    "tiffLocation"
+  ],
+  "properties": {
+    "greenBand": {
+      "$id": "#/properties/greenBand",
+      "type": "integer",
+      "title": "The Greenband Schema",
+      "default": 0,
+      "examples": [
+        3
+      ]
+    },
+    "redBand": {
+      "$id": "#/properties/redBand",
+      "type": "integer",
+      "title": "The Redband Schema",
+      "default": 0,
+      "examples": [
+        2
+      ]
+    },
+    "tiffLocation": {
+      "$id": "#/properties/tiffLocation",
+      "type": "string",
+      "title": "The Tifflocation Schema",
+      "default": "",
+      "examples": [
+        "s3://cool-bucket/image.tiff"
+      ],
+      "format": "uri"
+    }
+  }
+}
+```
+
+With this JSON schema attached to our model, it's impossible to submit
+improperly formatted arguments.
+
+Over time, your model's input requirements may change. In that case it makes
+sense to make a new model with a different validator. Since writing JSON schema
+by hand is tedious, it's helpful to use [`json-schema.net`](https://jsonschema.net/)
+to generate schemas from examples, then edit those to match your needs.
+
+## What's a `Prediction`?
+
+A `Prediction` is a single run of a model with specific inputs. Predictions are
+created when you `POST` a model id and arguments to `/api/predictions`.
+
+```scala mdoc
+val modelId = UUID.randomUUID
```

Comment:
I am not sure how `Prediction`s are being being connected to `Model`s. In this example, the `modelId` is random. I would expect it to be a well known ID returned from `POST /api/models`, but I may be completely misunderstanding relationships between the entities.

Reply:
I will move the model id out to a configured variable for `mdoc` and use a consistent one throughout

Reply:
This is done, and now the ids link up across the examples. I didn't understand the scope rules for my different `mdoc` blocks, but it turns out variables are accessible between them. Neat!

Reply:
It is done, now all the model IDs are the same